### ISSUE TITLE
Context help in VB.NET app does not work

### ISSUE THREAD

**Issue by MihaIvan** [EWSoftware/SHFB#891]
I added some pages to a help project that was previously working. I tried testing the changes, and the help form failed to show after using PF1. I tried it again, and got the arrow chasing its tail for a moment, the nothing. I shut down the program and tried looking at the .chm file by double clicking it. I get the message: Cannot open the file: mk:@MSITStore:... where ... is the path to the chm file.

If I copy it to the executable folder from the HELP folder in the help project, I can view the help. It is only after failing to view help from the app that I get the message.

I copied an unrelated chm file I generated in a different project to the executable folder. I renamed it to the name the program is expecting, and it is able to access the chm file, but gives the message: 
Can’t reach this page
Make sure the web address //ieframe.dll/dnserrordiagoff.htm# is correct 
Search for this site on Bing 
Refresh the page 

I rather expect this since the topic is not in that chm file. After I shut down the program, I can double click on the chm file and view it. I am not locked out. So it seems to me that there is something in the chm file that is not right.

I tried wiping out the help project altogether and just use a minimal file and that didn't work either, but I forgot to document the failure mode in that instance. 

Where do I go from here?

**Comment by EWSoftware** [comment]
Did you produce the help file with the Sandcastle Help File Builder?  I do know they work if invoked from an application.  Below is an example I just tried in LINQPad

```
Form form = new Form();
form.CreateControl();
Help.ShowHelp(form, @"C:\GH\SHFB\TestCaseProject\doc\Help\TestCaseProject.chm",
    HelpNavigator.Topic, "html/017f910a-5e5e-2e58-a60c-52b44deb0425.htm");
```

In the case of files produced by SHFB, the topics are stored in an *html* folder, are named after their generated topic IDs, and have a *.htm" extension.  As long as the path to the topic is correct, it will display.

**Comment by MihaIvan** [comment]
Eric. Sorry I have not responded. There was a death in my wife’s family and we had to leave right away to beat the storm hitting St. Louis. Then we got stuck in the ice storm in Memphis.

This is not how I am invoking help. I am using the HelpProvider class on a Window form. When the user uses the F1 key, it uses the HelpKeyword and HelpNavigator properties to show the help topic. It all happens under the covers.

**Comment by EWSoftware** [comment]
Did you build the help file with SHFB or something else?

**Comment by MihaIvan** [comment]
I build it in Visual Studio.

**Comment by EWSoftware** [comment]
I'm referring to the help file itself.  Did you build that with the Sandcastle Help File Builder extension (.shfbproj) or something else?

**Comment by MihaIvan** [comment]
I do everything from inside Visual Studio. I right click on the Sandcastle project in VS and click on build to build the .chm file as well as the website help.

**Comment by EWSoftware** [comment]
I've never used SHFB to produce context sensitive help for an application.  It should work though if you specify the right topic ID.  From your original post, it would appear you can't open the help file directly so that would indicate some sort of issue with the help file itself.  I'd suggest rebuilding the help file and seeing if you can just open it using the View Help option from with the SHFB project.

**Comment by MihaIvan** [comment]
I can double click on the .chm file, and it opens correctly. It is only when I try to access it as context sensitive help from a windows form.

**Comment by MihaIvan** [comment]
This makes no sense. I started out with GUI IDs as topics. Using those seems ridiculous to me because you cannot look at them and tell what you are looking at. So I changed them all to words. I changed the references in the forms to match by using a mass change. That is where my problems began.

I tried stripping down the help to only the context help for the opening form, and this still did not work. I substituted another .chm file and changed the topic ID in the form and that worked. That is when I got you involved.

So finally I scrapped everything in the help. I went back to GUIIDs and that worked for the first form context help. So I changed the topic ID along with the content layout stuff to words. This worked. Then I copied the old content to that new project. Built it from VS like I usually do, and everything work. So essentially I wound up back where I was when it did not work before.

I am sure I did something differently, but I have no idea what that was.

Sorry for stirring up the pot.
